lib/ansible/module_utils/facts/system/pkg_mgr.py

```diff
@@ -17,7 +17,13 @@
 # ansible module, use that as the value for the 'name' key.
 PKG_MGRS = [{'path': '/usr/bin/rpm-ostree', 'name': 'atomic_container'},
             {'path': '/usr/bin/yum', 'name': 'yum'},
-            {'path': '/usr/bin/dnf', 'name': 'dnf'},
```

Comment:
I'm not sure removing this is a good idea. /usr/bin/dnf5 will always be present if dnf5 is installed, but I don't think /usr/bin/dnf-3 is synonymous with dnf4. IMO, it would be better to keep this for distributions other than Fedora which we don't special case.

Reply:
The intent behind removing `/usr/bin/dnf` from `PKG_MGRS` is trying to avoid potential false positives since there is no guarantee what `/usr/bin/dnf` is at this point.

> I don't think /usr/bin/dnf-3 is synonymous with dnf4

Can you be a more specific? Are you saying that `/usr/bin/dnf-3` doesn't have to necessarily exist for `dnf != 5`?

> distributions other than Fedora which we don't special case

We don't just "special case" Fedora, we do all of this dnf heuristic for "derivatives of RHEL". OpenMandriva which doesn't belong to that group and uses dnf seems to have `/usr/bin/dnf-3` present on the filesystem. Are there others?

Reply:
Is `/usr/bin/dnf-2` actually used as a default package manager somewhere? We could add that to `PKG_MGR` just to be on the safe side. For `ansible_os_family == RedHat` the current code in the PR would still match `dnf` (dnf4) module for `/usr/bin/dnf-2` correctly.

Reply:
I'm not sure about the exact history (might be a good idea to ask jmracek or someone else on that team), but according to the Fedora 22 dnf specfile [1], /usr/bin/dnf-2 was the default in Fedora 22 and /usr/bin/dnf-3 was the default in Fedora 23 and thereafter. In earlier releases, yum, not dnf, was the default package manager. However, RedHat-ish distributions are already special cased, and I have no idea about other distributions, so take that point for what you will.

In any case, this shouldn't be a problem for recent versions, so I'd rather merge this as is than have a long, drawn out discussion :smiley:.


[1] https://src.fedoraproject.org/rpms/dnf/blob/f22/f/dnf.spec#_170